appinventor/components/src/com/google/appinventor/components/runtime/util/ListViewArrayAdapterImageSingleText.java

```diff
@@ -0,0 +1,142 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2019 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime.util;
+
+import android.graphics.drawable.Drawable;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import com.google.appinventor.components.runtime.ComponentContainer;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Adapter class to populate ListView with an image and single line of text
+ */
+public class ListViewArrayAdapterImageSingleText {
```

Comment:
The various adapter classes seem to share quite a bit of code. It might make sense to consider refactoring them so that there is a common base class with specializations for each different layout. I would recommend writing a paragraph or two in your GSOC final document about how you would do it, but don't consider it a priority for finishing your work.

Reply:
Vinay used a combined adapter class. I intend to use his when his work is merged in.